website/blog/2025-06-12-react-native-0.80.md

```diff
@@ -0,0 +1,225 @@
+---
+title: 'React Native 0.80 - React 19.1, JS API Changes, Freezing Legacy Arch and much more'
+authors: [hezi, fabriziocucci, gabrieldonadel, chrfalch]
+tags: [engineering]
+date: 2025-06-12
+---
+
+# React Native 0.80 - React 19.1, JS API Changes, Freezing Legacy Arch and much more
+
+Today we are excited to release React Native 0.80!
+
+This release brings the version of React we ship inside React Native to the latest stable available: 19.1.0.
+
+We’re also shipping a series of stability improvements to our JS API: deep imports will now fire a warning and we’re offering a new opt-in Strict TypeScript API which offers types that are more accurate and safer to use.
+
+Moreover, the Legacy Architecture of React Native is now officially frozen, and you’ll start seeing warnings for APIs that will stop working once we fully sunset the Legacy Architecture.
+
+### Highlights
+
+- [JavaScript deep imports deprecation](/blog/2025/06/12/react-native-0.80#javascript-deep-imports-deprecation)
+- [Legacy Architecture Freezing & Warnings](/blog/2025/06/12/react-native-0.80#legacy-architecture-freezing--warnings)
+- [React 19.1.0](/blog/2025/06/12/react-native-0.80#react-1910)
+- [Experimental - React Native iOS dependencies are now prebuilt](/blog/2025/06/12/react-native-0.80#experimental---react-native-ios-dependencies-are-now-prebuilt)
+
+<!--truncate-->
+
+## Highlights
+
+### JavaScript deep imports deprecation
+
+In this release, we are making moves to improve and stabilize React Native's public JavaScript API. The first step towards this is better scoping of which of our APIs are importable by apps and frameworks. In line with this, we are formally deprecating deep imports from React Native ([see RFC](https://github.com/react-native-community/discussions-and-proposals/pull/894)), and are introducing warnings via ESLint and the JS console.
+
+These warnings are scoped to imports from within your project's source code, and can be [opted out from](/docs/strict-typescript-api). However, please bear in mind that we aim to remove deep imports from React Native's API in a future release, and these should instead be updated to the root import.
+
```

Comment:
Can we break this section up / have a focus point with the illustration?

```suggestion
\```js title=""
// Before - import from subpath
import {Alert} from 'react-native/Libraries/Alert/Alert';

// After - import from `react-native`
import {Alert} from 'react-native';
\```
```